db/migrate/20200803113903_migrate_host_type_in_host_config_groups.foreman_puppet.rb

```diff
@@ -51,4 +33,38 @@ def down
                          .update_all(host_type: 'Hostgroup', host_id: hostgroup_facet_ids[hostgroup_facet_id])
     end
   end
+
+  private
+
+  def migrate_host_data_to_facets
+    host_config_group_ids = FakeHostConfigGroup.for_host.pluck(:host_id).uniq
+    host_config_group_ids.each do |host_id|
+      host_facet_id = ForemanPuppet::HostPuppetFacet.where(host_id: host_id).pick(:id)
+      host_groups = FakeHostConfigGroup.for_host.where(host_id: host_id)
+      if host_facet_id
+        host_groups.update_all(host_type: 'ForemanPuppet::HostPuppetFacet', host_id: host_facet_id)
+      else
+        deleted_groups = FakeConfigGroup.where(id: host_groups.pluck(:config_group_id)).pluck(:name)
+        say "deleting groups #{deleted_groups.join(', ')} from Host (id=#{host_id}) because it doesn't exists or has no environment"
```

Comment:
```suggestion
        say "Deleting config groups #{deleted_groups.join(', ')} from Host (id=#{host_id}) because it does not exist."
```

Reply:
You don't check for environment, do you? And what does `say` do? Why not using a logger? The string is not marked for translation, either.

Reply:
Say is used in migrations and uses the migration logger. It's a system log so I would not translate.

Reply:
What to check environment for? :thinking: